skrifa/src/outline/cff/mod.rs

```diff
@@ -199,34 +199,43 @@ impl<'a> Outlines<'a> {
         let charstring_data = charstrings.get(glyph_id.to_u32() as usize)?;
         let subrs = subfont.subrs(self)?;
         let blend_state = subfont.blend_state(self, coords)?;
+        let cs_eval = CharstringEvaluator {
+            cff_data,
+            charstrings,
+            global_subrs: self.global_subrs.clone(),
+            subrs,
+            blend_state,
+            charstring_data,
+        };
+        let is_scaled = subfont.scale.is_some();
         let mut pen_sink = PenSink::new(pen);
         let mut simplifying_adapter = NopFilteringSink::new(&mut pen_sink);
-        // Only apply hinting if we have a scale
-        if hint && subfont.scale.is_some() {
-            let mut hinting_adapter =
-                HintingSink::new(&subfont.hint_state, &mut simplifying_adapter);
-            charstring::evaluate(
-                cff_data,
-                charstrings,
-                self.global_subrs.clone(),
-                subrs,
-                blend_state,
-                charstring_data,
-                &mut hinting_adapter,
-            )?;
-            hinting_adapter.finish();
+        if let Some(matrix) = self.top_dict.font_matrix {
```

Comment:
Optional: Between this branch and the else branch there seems to be some overlap. I don't immediately see a waz, but i am wondering if this could be compacted / folded together a bit more?

Reply:
The static type signatures are different for all of the branches so unfortunately there’s no way to coalesce these. I did extract the evaluation code out in attempt the compact this a bit.